### PR TITLE
logging corrections and minor improvements

### PR DESCRIPTION
**Pull request by taroface**
- Fixed some YAML alignment and link formatting.
- Clarified how the YAML defaults are overriden.
- Moved `DEV` channel section to a more intuitive location on the page.

### CODE REVIEW

**Comment by taroface** [comment]
@mikeCRL This PR isn't responding to any request, but I was reading over the logging docs and thought they could use some improvement. Would you mind taking a look (not urgent)?

File: v21.1/configure-logs.md [hunk 1]
```diff
@@ -425,5 +422,5 @@ Note that a default `dir` is not specified for `file-defaults` and `capture-stra
 ## See also
 
 - [Logging Use Cases](logging-use-cases.html)
-- [log sinks](log-sinks.html)
+- [Log Sinks](log-sinks.html)
 - [Log formats](log-formats.html)
```
Comment (mikeCRL) [inline v21.1/configure-logs.md:426]:
```suggestion
- [Log Formats](log-formats.html)
```

Caps to match that page title and the other items here

**Review by mikeCRL — APPROVED** [review]
LGTM! Suggested one character be changed to caps.

**Review by taroface — COMMENTED** [review]
TFTR!!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10570#-:-Ma0OXz9666snk8XUY98:bz5m8k5)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mikeCRL)

___
*[v21.1/configure-logs.md, line 426 at r1](https://reviewable.io/reviews/cockroachdb/docs/10570#-M_cZKa4emUK36wpiHo3-r1-426:-Ma0OXBa15OEbtqJqCgM:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/1e3068c0d8498a3178bd878dc82603102588d52d/v21.1/configure-logs.md#L426)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- [Log Formats](log-formats.html)
```

Caps to match that page title and the other items here
</blockquote></details>

Done.

**Pull request merged**